### PR TITLE
Connect to HTTP datasource

### PR DESCRIPTION
**Pull request by chrisbonifacio**
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://docs.amplify.aws/)` 
            HTML: `<a href="https://docs.amplify.aws/">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by aspittel — APPROVED** [review]

**Comment by reesscot** [comment]
@chrisbonifacio I updated this to point to `mai`n and updated with latest link fixes from `main`

**Review by reesscot — APPROVED** [review]

**Comment by chrisbonifacio** [comment]
> @chrisbonifacio I updated this to point to `mai`n and updated with latest link fixes from `main`

Thank you, Scott!

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-http-datasource/index.mdx [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+  authorizationModes: {
+    defaultAuthorizationMode: 'apiKey',
+    apiKeyAuthorizationMode: {
+      expiresInDays: 30,
+    },
+  },
+});
+```
+
+## Step 2 - Add your REST API as Datasource
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-http-datasource/index.mdx:75]:
```suggestion
## Step 2 - Add your REST API/HTTP API as Datasource
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-http-datasource/index.mdx [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+
+To integrate the external REST API, you'll need to set it up as the HTTP Datasource. Add the following code in your `amplify/backend.ts` file.
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-http-datasource/index.mdx:77]:
```suggestion
To integrate the external REST API or HTTP API, you'll need to set it up as the HTTP Datasource. Add the following code in your `amplify/backend.ts` file.
```

File: src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-http-datasource/index.mdx [hunk 3]
```diff
@@ -0,0 +1,440 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Connect to HTTP Datasource',
```
Comment (renebrandel) [inline src/pages/[platform]/build-a-backend/data/custom-business-logic/connect-http-datasource/index.mdx:4]:
```suggestion
  title: 'Connect to an external HTTP endpoint',
```

**Review by renebrandel — APPROVED** [review]

**Review by cwomack — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**